pkg/api/v1beta3/types.go

```diff
@@ -995,6 +1006,16 @@ type Binding struct {
 	Target ObjectReference `json:"target" description:"an object to bind to"`
 }
 
+// DeleteOptions may be provided when deleting an API object
+type DeleteOptions struct {
+	TypeMeta `json:",inline"`
```

Comment:
This is kind and version of the object being deleted? But not namespace, name, etc.? Why are the kind and version needed? I suppose even Status has TypeMeta and ListMeta, but I don't think we need to be able to serialize DeleteOptions, such as in config files.

Actually, it would be nice to at least be able to specify uid and resourceVersion, to facilitate preconditions.